rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -454,6 +454,13 @@ rmw_connextdds_get_readerwriter_qos(
   }
 #endif /* RMW_CONNEXT_DDS_API == RMW_CONNEXT_DDS_API_PRO */
 
+  auto user_data_str = rmw_dds_common::encode_type_hash_for_user_data_qos(
+    *type_support->type_hash());
```

Comment:
This needs to be updated with the new API and do error checking.  Also, let's not use `auto` here.